integrationTests/multiShard/hardFork/hardFork_test.go

```diff
@@ -74,18 +74,23 @@ func TestEpochStartChangeWithoutTransactionInMultiShardedEnvironment(t *testing.
 
 	time.Sleep(time.Second)
 
-	_, _ = integrationTests.WaitOperationToBeDone(t, nodes, nrRoundsToPropagateMultiShard, nonce, round, idxProposers)
+	nonce, _ = integrationTests.WaitOperationToBeDone(t, nodes, nrRoundsToPropagateMultiShard, nonce, round, idxProposers)
 
 	time.Sleep(time.Second)
 
 	epoch := uint32(1)
 	verifyIfNodesHaveCorrectEpoch(t, epoch, nodes)
-	verifyIfNodesHaveCorrectNonce(t, nonce, nodes)
+	verifyIfNodesHaveCorrectNonce(t, nonce-1, nodes)
 	verifyIfAddedShardHeadersAreWithNewEpoch(t, nodes)
 
-	createHardForkExporter(t, nodes)
+	defer func() {
```

Comment:
I think we might not use physical files in integration tests. TestProcessorNodes use memory storage units.

Reply:
MultiFileWriter has to write directly in files, it has the code to open folder and open files